sound/soc/sof/intel/hda-loader.c

```diff
@@ -167,8 +167,6 @@ static int cl_trigger(struct snd_sof_dev *sdev,
 	/* code loader is special case that reuses stream ops */
 	switch (cmd) {
 	case SNDRV_PCM_TRIGGER_START:
-		wait_event_timeout(sdev->waitq, !sdev->code_loading,
-				   HDA_DSP_CL_TRIGGER_TIMEOUT);
```

Comment:
why would we remove a timeout? this is a safe practice, if you remove it when how will be know if the code loading fails?

Reply:
@plbossart I think the code_loading here was misused, at this point the FW is not downloaded yet, we need "trigger start" the DMA to make the downloading initiated. So here what we were waiting is actually the "PURGE FW" done reply, but we already had timeout polling at the init() which has already verified about that, so the waitq and code_loading check make no sense, only adding confusion.